src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2058,57 +2058,81 @@ void CodeGen::genFMAIntrinsic(GenTreeHWIntrinsic* node)
     // Intrinsics with CopyUpperBits semantics cannot have op1 be contained
     assert(!copiesUpperBits || !op1->isContained());
 
+    // We need to keep this in sync with lsraxarch.cpp
+    // Ideally we'd actually swap the operands in lsra and simplify codegen
+    // but its a bit more complicated to do so for many operands as well
+    // as being complicated to tell codegen how to pick the right instruction
+
+    instruction ins = INS_invalid;
+
     if (op1->isContained() || op1->isUsedFromSpillTemp())
     {
+        // targetReg == op3NodeReg or targetReg == ?
+        // op3 = ([op1] * op2) + op3
+        // 231 form: XMM1 = (XMM2 * [XMM3]) + XMM1
+        ins = _231form;
+        std::swap(emitOp1, emitOp3);
+
         if (targetReg == op2NodeReg)
         {
-            std::swap(emitOp1, emitOp2);
             // op2 = ([op1] * op2) + op3
             // 132 form: XMM1 = (XMM1 * [XMM3]) + XMM2
             ins = _132form;
             std::swap(emitOp2, emitOp3);
         }
```

Comment:
After ```std::swap(emitOp1, emitOp3);``` and ```std::swap(emitOp2, emitOp3);```, we will get ```emitOp1 = op3; emitOp2 = op1; emitOp3 = op2```. But in my understanding, we need ``` emitOp1 = op2; emitOp2 = op3; emitOp3 = op1 ``` here. Is there anything I'm missing?

Reply:
@tannergooding, in case you haven't seen this one #62343. That was a case missed and captured by pmi.

Reply:
You're right and that means LSRA has an issue here. Will fix